### PR TITLE
Cyclopedia -> Character

### PR DESCRIPTION
**Pull request by dex-89**
**I tested. It doesn't crash the client or the server. Works perfectly for me.**

**Before:**
2020-11-10_175247414_GOD_Hotkey

**After:**
2020-11-10_182556789_GOD_Hotkey

Compilation @ my Debian server:
screen

**Source code:** source from this [https://github.com/malucooo/otxserver-new]() project

### CODE REVIEW

**Review by CyCu991 — COMMENTED** [review]
Working fine +5 for u :)

**Comment by dex-89** [comment]
> [ 1%] Checking the git repository for changes...
> [ 1%] Built target check_git
> [ 3%] Building CXX object CMakeFiles/otbr_lib.dir/src/otpch.cpp.o
> [ 3%] Building CXX object CMakeFiles/otbr_lib.dir/src/account.cpp.o
> [ 4%] Building CXX object CMakeFiles/otbr_lib.dir/src/actions.cpp.o
> [ 5%] Building CXX object CMakeFiles/otbr_lib.dir/src/ban.cpp.o
> In file included from /home/ubuntu/otspl-global/src/actions.cpp:26:
> /home/ubuntu/otspl-global/src/game.h:347:57: error: ‘CyclopediaCharacterInfoType_t’ has not been declared
> 347 | void playerCyclopediaCharacterInfo(uint32_t playerId, CyclopediaCharacterInfoType_t characterInfoType);
> | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> [ 6%] Building CXX object CMakeFiles/otbr_lib.dir/src/baseevents.cpp.o
> make[2]: *** [CMakeFiles/otbr_lib.dir/build.make:89: CMakeFiles/otbr_lib.dir/src/actions.cpp.o] Error 1
> make[2]: *** Waiting for unfinished jobs....
> make[1]: *** [CMakeFiles/Makefile2:155: CMakeFiles/otbr_lib.dir/all] Error 2
> make: *** [Makefile:84: all] Error 2

You have somethings wrong. I compile it with no errors

Try download [https://github.com/dex-89/otservbr-global/tree/patch-1]() and compile it

File: src/protocolgame.cpp [hunk 1]
```diff
@@ -1561,6 +1593,332 @@ void ProtocolGame::sendAddMarker(const Position& pos, uint8_t markType, const st
...
+	NetworkMessage msg;
+	msg.reset();
+	msg.addByte(0xDA);
+	msg.addByte(CYCLOPEDIA_CHARACTERINFO_GENERALSTATS);
+	msg.addByte(0x00); // No data available
+	msg.add<uint64_t>(player->getExperience());
+	msg.add<uint16_t>(player->getLevel());
+	msg.addByte(player->getLevelPercent());
+	msg.add<uint16_t>(100); // base xp gain rate
+	if (version >= 1215) {
```
Comment (FakeShinoda) [inline src/protocolgame.cpp]: Please remove the "if (version)" we use only one version.

File: src/protocolgame.cpp [hunk 2]
```diff
@@ -1561,6 +1593,332 @@ void ProtocolGame::sendAddMarker(const Position& pos, uint8_t markType, const st
...
+	msg.reset();
+	msg.addByte(0xDA);
+	msg.addByte(CYCLOPEDIA_CHARACTERINFO_COMBATSTATS);
+	msg.addByte(0x00); // No data available
+	for (uint8_t i = SKILL_CRITICAL_HIT_CHANCE; i <= SKILL_LAST; ++i) {
+		msg.add<uint16_t>(std::min<int32_t>(player->getSkillLevel(i), std::numeric_limits<uint16_t>::max()));
+		msg.add<uint16_t>(0);
+	}
+	uint8_t haveBlesses = 0;
+	uint8_t blessings = (version >= 1130 ? 8 : 6);
```
Comment (FakeShinoda) [inline src/protocolgame.cpp]: uint8_t blessings = 8;

File: src/protocolgame.cpp [hunk 3]
```diff
@@ -1561,6 +1593,332 @@ void ProtocolGame::sendAddMarker(const Position& pos, uint8_t markType, const st
...
+	}
+	else {
+		msg.addByte(player->getMagicLevelPercent());
+	}
+	for (uint8_t i = SKILL_FIRST; i < SKILL_CRITICAL_HIT_CHANCE; ++i) { //TODO: check if all clients have the same hardcoded skill ids
+		static const uint8_t HardcodedSkillIds[] = { 11, 9, 8, 10, 7, 6, 13 };
+		msg.addByte(HardcodedSkillIds[i]);
+		msg.add<uint16_t>(std::min<int32_t>(player->getSkillLevel(i), std::numeric_limits<uint16_t>::max()));
+		msg.add<uint16_t>(player->getBaseSkill(i));
+		if (version >= 1200) {
```
Comment (FakeShinoda) [inline src/protocolgame.cpp]: Same here

**Review by FakeShinoda — CHANGES_REQUESTED** [review]

File: src/protocolgame.cpp (hunk 3 above)
Comment (dex-89) [reply src/protocolgame.cpp]: I forgot about that. Thanks

**Review by FakeShinoda — APPROVED** [review]

**Comment by FakeShinoda** [comment]
LGTM. Thanks for your contribution. <3

**Comment by dex-89** [comment]
@FakeShinoda Im already working on a new element: highscores :) but i must test it

**Pull request merged**